django/tasks/backends/base.py

```diff
@@ -0,0 +1,116 @@
+
+        If the task defines it, use that, otherwise, fall back to the backend.
+        """
+        # If this project doesn't use a database, there's nothing to commit to
+        if not connections.settings:
+            return False
+
+        if task.enqueue_on_commit is not None:
+            return task.enqueue_on_commit
+
+        return self.enqueue_on_commit
+
+    def validate_task(self, task):
+        """
+        Determine whether the provided task is one which can be executed by the backend.
+        """
+        if not is_module_level_function(task.func):
+            raise InvalidTaskError("Task function must be defined at a module level")
+
+        if not self.supports_async_task and iscoroutinefunction(task.func):
+            raise InvalidTaskError("Backend does not support async tasks")
+
+        if (
+            task.priority < MIN_PRIORITY
+            or task.priority > MAX_PRIORITY
+            or int(task.priority) != task.priority
+        ):
+            raise InvalidTaskError(
+                f"priority must be a whole number between {MIN_PRIORITY} and "
+                f"{MAX_PRIORITY}"
+            )
+
+        if not self.supports_defer and task.run_after is not None:
+            raise InvalidTaskError("Backend does not support run_after")
+
+        if task.run_after is not None and not timezone.is_aware(task.run_after):
+            raise InvalidTaskError("run_after must be an aware datetime")
+
+        if self.queues and task.queue_name not in self.queues:
+            raise InvalidTaskError(
+                f"Queue '{task.queue_name}' is not valid for backend"
+            )
+
+    @abstractmethod
+    def enqueue(self, task, args, kwargs):
+        """
+        Queue up a task to be executed
+        """
+        ...
+
+    async def aenqueue(self, task, args, kwargs):
+        """
+        Queue up a task function (or coroutine) to be executed
+        """
+        return await sync_to_async(self.enqueue, thread_sensitive=True)(
+            task=task, args=args, kwargs=kwargs
+        )
+
+    def get_result(self, result_id):
+        """
+        Retrieve a result by its id (if one exists).
+        If one doesn't, raises ResultDoesNotExist.
+        """
+        raise NotImplementedError(
+            "This backend does not support retrieving or refreshing results."
+        )
+
+    async def aget_result(self, result_id):
+        """
+        Queue up a task function (or coroutine) to be executed
+        """
+        return await sync_to_async(self.get_result, thread_sensitive=True)(
+            result_id=result_id
+        )
+
+    def check(self, **kwargs):
+        if self.enqueue_on_commit and not connections.settings:
+            yield messages.CheckMessage(
+                messages.ERROR,
+                "`ENQUEUE_ON_COMMIT` cannot be used when no databases are configured",
+                hint="Set `ENQUEUE_ON_COMMIT` to False",
+            )
```

Comment:
I think you should use `messages.Error` (I would also import Error directly) and the check should be [documented](https://docs.djangoproject.com/en/5.1/ref/checks/#core-system-checks)
It's also not tested

Reply:
I'd love some help in testing this. Specifically, a nice way to overwrite `connections.settings`, since `settings.DATABASES` isn't really configurable mid-test.

I guess `unittest.mock` could work?

Reply:
Hey @RealOrangeOne -- in this recent [comment](https://github.com/django/django/pull/18949#discussion_r1912530854) Simon discovered a nifty way of mocking `django.db.connections.__dict__`, does that help at all?

Reply:
Ooh interesting - I'll give it a go!

Reply:
A custom connection handler injected with a `mock.patch` does indeed work perfectly :partying_face: